yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -809,7 +809,13 @@ object Client extends Logging {
       }
     }
     addFileToClasspath(new URI(sparkJar(sparkConf)), SPARK_JAR, env)
-    populateHadoopClasspath(conf, env)
+    // Since we have a spark assembly that is including all the yarn and other dependencies we need,
+    // add an option to allow the user to not include the cluster default yarn/mapreduce application
+    // classpaths when running spark on yarn.
+    val isPopulateHadoopClasspath = conf.getBoolean("spark.yarn.cluster.classpath.populate", true)
```

Comment:
This name seems a little weird to me.  In particular, periods should be used to separate config namespaces, not in places where we'd use spaces in English.  I'd go with something like `spark.yarn.includeClusterHadoopJars` or `spark.yarn.includeClusterHadoopClasspath`.